### PR TITLE
Akka.Cluster MultiNode specs for node restarts, downing, and rejoining

### LINKED ISSUE #1821: Port Akka.Cluster.Tests.MultiNode.NodeRestartSpec
Biggest source of issues and bugs with Akka.Cluster, and should be able to isolate the issues with #1700. I've ported most of the spec and am working through the bugs it has exposed now.

### LINKED ISSUE #1830: Port Akka.Cluster.Tests.MultiNode.UnreachableNodeJoinsAgainSpec
Necessary for verifying some of the bugs related to #1700

### PR DESCRIPTION
**Pull request by Aaronontheweb**
close #1821 - Akka.Cluster.Tests.MultiNode.RestartNodeSpec added and passes
close #1830 - Akka.Cluster.Tests.MultiNode.UnreachableNodeJoinsAgainSpec added and passes

### CODE REVIEW

File: src/core/Akka.MultiNodeTestRunner/Program.cs [hunk 1]
```diff
@@ -132,7 +132,7 @@ static void Main(string[] args)
                             continue;
                         }
 
-                        if (!string.IsNullOrWhiteSpace(specName) && !test.Value.First().MethodName.Contains(specName))
+                        if (!string.IsNullOrWhiteSpace(specName) && !test.Value[0].MethodName.Contains(specName))
```
Comment (Aaronontheweb) [inline src/core/Akka.MultiNodeTestRunner/Program.cs]: Added this change to make it easier to debug with watches when I'm launching the MNTR

File: src/core/Akka.Remote.TestKit/BarrierCoordinator.cs [hunk 2]
```diff
@@ -244,7 +244,7 @@ public override int GetHashCode()
         public sealed class WrongBarrierException : Exception
         {
             public WrongBarrierException(string barrier, IActorRef client, Data barrierData)
-                : base(string.Format("tried"))
+                : base($"tried to enter '{barrier}' while we were waiting for '{barrierData.Barrier}'")
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote.TestKit/BarrierCoordinator.cs]: Filled out the rest of the error message

File: src/core/Akka.Remote.TestKit/Internals/Reference.conf [hunk 3]
```diff
@@ -25,7 +25,7 @@ akka {
     connect-timeout = 20s
     
     # Number of connect attempts to be made to the conductor controller
-    client-reconnects = 10
+    client-reconnects = 30
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote.TestKit/Internals/Reference.conf]: Made consistent with JVM

File: src/core/Akka.Remote.TestKit/Player.cs [hunk 4]
```diff
@@ -328,7 +328,7 @@ public void InitFSM()
                 }
                 if (@event.FsmEvent is StateTimeout)
                 {
-                    _log.Error("connect timeout to TestConductor");
+                    _log.Error($"Failed to connect to test conductor within {_settings.ConnectTimeout.TotalMilliseconds} ms.");
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote.TestKit/Player.cs]: More detailed error message

**Comment by annymsMthd** [comment]
Looks good to me

**Pull request merged**